src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ConnectedDevicesSpec.groovy

```diff
@@ -1383,6 +1384,174 @@ srcDevices=#newSrcEnabled, dstDevices=#newDstEnabled"() {
+        assumeTrue(topology.activeTraffGens.size() > 0, "Require at least 1 switch with connected traffgen")
+        def sw = topology.activeTraffGens[0].switchConnected
+        def initialPropsSource = enableMultiTableIfNeeded(true, sw.dpId)
+        switchHelper.updateSwitchProperties(sw, initialPropsSource.jacksonCopy().tap {
+            it.multiTable = true
+            it.switchLldp = false
+            it.switchArp = false
+        })
+
+        and: "Flow is created on a target switch with devices feature 'off'"
+        def dst = topology.activeSwitches.find { it.dpId != sw.dpId }
+        def initialPropsDst = enableMultiTableIfNeeded(true, dst.dpId)
+        def flow = flowHelperV2.randomFlow(sw, dst)
+
+        def outerVlan = 100
+        flow.source.vlanId = outerVlan
+        flow.source.innerVlanId = 200
+
+        flow.source.detectConnectedDevices = new DetectConnectedDevicesV2(false, false)
+        flow.destination.detectConnectedDevices = new DetectConnectedDevicesV2(false, false)
+        flowHelperV2.addFlow(flow)
+        northboundV2.getConnectedDevices(flow.source.switchId).ports.empty
+        northboundV2.getConnectedDevices(flow.destination.switchId).ports.empty
+
+        when: "update device properties and send lldp and arp packets on src flow endpoint and match outerVlan only"
+        switchHelper.updateSwitchProperties(sw, initialPropsSource.jacksonCopy().tap {
+            it.multiTable = true
+            it.switchLldp = true
+            it.switchArp = true
+        })
+
+        def srcLldpData = LldpData.buildRandom()
+        def dstLldpData = LldpData.buildRandom()
+        def srcArpData = ArpData.buildRandom()
+        def dstArpData = ArpData.buildRandom()
+        [[flow.source, srcLldpData, srcArpData], [flow.destination, dstLldpData, dstArpData]].each {
+            endpoint, lldpData, arpData ->
+                new ConnectedDevice(tgService, topology.getActiveTraffGen(endpoint.switchId), [outerVlan]).withCloseable {
+                    it.sendLldp(lldpData)
+                    it.sendArp(arpData)
+                }
+        }
+
+        then: "Getting connecting devices doesn't show corresponding devices on src endpoint"
+        def response = northboundV2.getConnectedDevices(sw.dpId)
+        sleep(3000)
+        Wrappers.timedLoop(3) {
+            //under usual condition system needs some time for devices to appear, that's why timeLoop is used here
+            verifyAll(northboundV2.getConnectedDevices(sw.dpId)) {
+                !(it.ports.lldp.empty)
+            }
+        }
+
+        cleanup: "Delete the flow and restore initial switch properties"
+        flow && flowHelperV2.deleteFlow(flow.flowId)
+        initialPropsSource && restoreSwitchProperties(sw.dpId, initialPropsSource)
+        initialPropsDst && restoreSwitchProperties(dst.dpId, initialPropsDst)
+    }
+
+    @Tidy
+    def "System stops receiving statistics if config is changed to 'off'"() {
```

Comment:
You may consider merging this test with the `Switch is not containing extra rules after connection devices removal`, since they have similar test actions and only verifications are different. It's okay to leave them as separate if you think that that makes sense

Reply:
yes, i would like to leave separate tests to avoid one of the checks failures (not run) another if it comes after

Reply:
Sure it's your call. This scenario does not look too long to execute, so it doesn't really matter. But test run time is also a very important metric. At some point you will have to drop some tests from 'smoke' suite to 'regression' suite in order to get feedback in a reasonable time (previously we tried to stay within 1h for smoke tests, not counting env boot time). To get around the problem that some verifications may not run and you really want them to run, take a look at the usage of SoftAssertions class around the project, it will allow to delay the test failure until all required verifications happen